include/tracing_impl/lttng.h

```diff
@@ -2,11 +2,6 @@
  * Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All rights reserved.
  */
 
-#ifndef LTTNG_H_
-#define LTTNG_H_
-
-#if HAVE_LTTNG_UST
-
 #undef LTTNG_UST_TRACEPOINT_PROVIDER
```

Comment:
Not sure I understand this change. The header guard is so that our project includes this header just once. Unless we were conflicting with a LTTNG-level guard, what requirement does lttng have on us?

Reply:
LTTNG requires that the header be included multiple times in some cases. Here is a template of how the header should look: https://lttng.org/man/3/lttng-ust/v2.13/#doc-creating-tp
```
#undef LTTNG_UST_TRACEPOINT_PROVIDER
#define LTTNG_UST_TRACEPOINT_PROVIDER my_provider

#undef LTTNG_UST_TRACEPOINT_INCLUDE
#define LTTNG_UST_TRACEPOINT_INCLUDE "./tp.h"

#if !defined(_TP_H) || \
    defined(LTTNG_UST_TRACEPOINT_HEADER_MULTI_READ)
#define _TP_H

#include <lttng/tracepoint.h>

/*
 * LTTNG_UST_TRACEPOINT_EVENT(), LTTNG_UST_TRACEPOINT_EVENT_CLASS(),
 * LTTNG_UST_TRACEPOINT_EVENT_INSTANCE(),
 * LTTNG_UST_TRACEPOINT_LOGLEVEL(), and `LTTNG_UST_TRACEPOINT_ENUM()`
 * are used here.
 */

#endif /* _TP_H */

#include <lttng/tracepoint-event.h>
```

Basically `LTTNG_UST_TRACEPOINT_HEADER_MULTI_READ` should override the header guard check.

Reply:
This should go into the commit message.

Reply:
Done